lib/pbench/test/unit/server/database/test_api_keys_db.py

```diff
@@ -31,29 +26,44 @@ def fake_db(self, monkeypatch, server_config):
     def test_construct(self, db_session, create_drb_user, create_user):
         """Test api_key constructor"""
 
-        api_key = APIKey(
-            api_key="generated_api_key",
-            user=create_user,
-        )
+        api_key = APIKey(key="generated_api_key", user=create_user, name="test_api_key")
         api_key.add()
 
-        assert api_key.api_key == "generated_api_key"
+        assert api_key.key == "generated_api_key"
         assert api_key.user.id is create_user.id
+        assert api_key.name == "test_api_key"
+
+    def test_query(
+        self,
+        db_session,
+        pbench_drb_api_key,
+        pbench_drb_secondary_api_key,
+        create_drb_user,
+    ):
+        """Test that we are able to query api_key by user"""
 
-    def test_query(self, db_session, pbench_drb_api_key):
-        """Test that we can able to query api_key in the table"""
+        key_list = APIKey.query(user=create_drb_user)
+        assert len(key_list) == 2
 
-        key = APIKey.query(pbench_drb_api_key)
-        assert key.api_key == pbench_drb_api_key
-        assert key.user.id == DRB_USER_ID
-        assert key.user.username == "drb"
+        assert key_list[0].user.id == pbench_drb_api_key.user.id
+        assert key_list[0].name == pbench_drb_api_key.name
+        assert key_list[0].id == pbench_drb_api_key.id
+        assert key_list[0].key == pbench_drb_api_key.key
+        assert key_list[1].user.id == pbench_drb_secondary_api_key.user.id
+        assert key_list[1].name == pbench_drb_secondary_api_key.name
+        assert key_list[1].id == pbench_drb_secondary_api_key.id
+        assert key_list[1].key == pbench_drb_secondary_api_key.key
```

Comment:
This test now shows that we can find all keys which belong to a specific user (which is good), but it doesn't show that we _don't_ find keys which belong to **another** user.